sklearn/preprocessing/mice.py

```diff
@@ -0,0 +1,365 @@
+        n_features = X_filled.shape[1]
+        if self.n_nearest_columns > n_features - 1:
+            return None
+        abs_correlation_matrix = np.abs(np.corrcoef(X_filled.T))
+        # np.corrcoef is not defined for constant columns
+        abs_correlation_matrix[np.isnan(abs_correlation_matrix)] = eps
+        np.fill_diagonal(abs_correlation_matrix, 0)
+        abs_correlation_matrix = normalize(abs_correlation_matrix, norm='l1',
+                                           axis=0)
+        return abs_correlation_matrix
+
+    def fit_transform(self, X, y=None):
+        """
+        Fit the imputer on X and return the transformed X.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Input data, where "n_samples" is the number of samples and
+            "n_features" is the number of features.
+        """
+        X = check_array(X, dtype=np.float64, force_all_finite=False)
+        X = np.asarray(X, order="F")
+        mask_missing_values = _get_mask(X, self.missing_values)
+
+        # initial imputation
+        self.initial_imputer_ = Imputer(missing_values=self.missing_values,
+                                        strategy=self.initial_fill_method,
+                                        axis=0)
+        X_filled = self.initial_imputer_.fit_transform(X)
+        self._val_inds = self.initial_imputer_._valid_statistics_indexes
+        X = X[:, self._val_inds]
+        mask_missing_values = mask_missing_values[:, self._val_inds]
+
+        # perform imputations
+        n_samples, n_features = X_filled.shape
+        total_rounds = self.n_burn_in + self.n_imputations
+        results_list = []
+        if self.verbose:
+            print("[MICE] Completing matrix with shape %s" % (X.shape,))
+            start_t = time()
+            mice_msg = '[MICE] Ending imputation round '
+        for m in range(total_rounds):
+            # order in which to impute
+            ordered_indices = self._get_ordered_indices(mask_missing_values)
+
+            # abs_correlation matrix is used to choose a subset of other
+            # features to impute from  f
+            abs_corr_mat = self._get_abs_correlation_matrix(X_filled)
+
+            # Fill in each column in the order of ordered_indices
+            for this_column in ordered_indices:
+                other_columns = self._get_other_columns(n_features,
+                                                        this_column,
+                                                        abs_corr_mat)
+                X_filled, model = self._fill_in_one_column(X_filled,
+                                                           mask_missing_values,
+                                                           this_column,
+                                                           other_columns)
+                model_triplet = (this_column, other_columns, model)
+                self.trained_model_triplets.append(model_triplet)
```

Comment:
Should intialize this ariable at each transform? I still did not check how the algorithm is working